packages/video_player/tizen/src/video_player.cc

```diff
@@ -423,13 +450,84 @@ void VideoPlayer::OnError(int code, void *data) {
 
 void VideoPlayer::OnVideoFrameDecoded(media_packet_h packet, void *data) {
   auto *player = reinterpret_cast<VideoPlayer *>(data);
-
   std::lock_guard<std::mutex> lock(player->mutex_);
-  if (player->current_media_packet_) {
-    LOG_INFO("[VideoPlayer] Media packet already pending.");
-    media_packet_destroy(packet);
+  player->SendMessage(kMessageOnFrameDecoded, packet);
+}
+
+void VideoPlayer::RunMediaPacketLoop(void *data, Ecore_Thread *thread) {
+  auto *self = reinterpret_cast<VideoPlayer *>(data);
+  Eina_Thread_Queue *packet_thread_queue = eina_thread_queue_new();
+  if (!packet_thread_queue) {
+    LOG_ERROR("Invalid packet queue.");
+    ecore_thread_cancel(thread);
     return;
   }
-  player->current_media_packet_ = packet;
-  player->texture_registrar_->MarkTextureFrameAvailable(player->texture_id_);
+
+  self->packet_thread_queue_ = packet_thread_queue;
+  std::queue<media_packet_h> packet_queue;
+  while (!ecore_thread_check(thread)) {
+    void *ref;
+    Message *message = static_cast<Message *>(
+        eina_thread_queue_wait(packet_thread_queue, &ref));
+    if (message->event == kMessageQuit) {
+      LOG_ERROR("Message quit.");
```

Comment:
This isn't an error and may be removed.

Reply:
Did you decide to leave the `LOG_ERROR` message? I've suggested it because it's an expected flow of the program, not an error or exception.

Reply:
Ok, I change to INFO